### ISSUE TITLE
status of group download

### ISSUE THREAD

**Issue by duncanokeyo** [tonyofrancis/Fetch#198]
could you add the function to get the download status of group downloads.
 and could we have callbacks that report group download progress.

for example 

```
override fun pauseGroup(id: Int): Fetch {
        synchronized(lock) {
            throwExceptionIfClosed()
            handlerWrapper.post {
                try {
                    val downloads = fetchHandler.pausedGroup(id)
                    uiHandler.post {
                        downloads.forEach {
                            logger.d("Paused download $it")
                            listenerCoordinator.mainListener.onPaused(it)
                        }
                    }
                } catch (e: FetchException) {
                    logger.e("Fetch with namespace $namespace error", e)
                }
            }
            return this
        }
    }
```

instead of calling onpause callback for each download, we can simple call onpause for group once we are sure all the downloads in the group have been paused.

**Comment by duncanokeyo** [comment]
and could we have callbacks that report group download progress.

**Comment by tonyofrancis** [comment]
@duncanokeyo 
1. Fetch has methods `fetch.getDownloadsInGroup(groupId, callback)` and `fetch.getDownloadsInGroupWithStatus(groupid, status, callback)`

2. The reason why the on pauseGroup method calls the `listener.onPause(download)` method for each download in the group is because some downloads in the group may have different statuses. For example we cannot pause a download that has a status of completed. So the callback is only called for downloads that were paused. I the next version of Fetch the pauseGroup method will take in an optional callback that will return all the paused downloads.

**Comment by tonyofrancis** [comment]
@duncanokeyo  Just published Fetch 2.2.0-RC1. In addtion to my comment above. Fetch now has new methods that take 2 optional callbacks. The changelog is huge so you may want to review it. Looking forward to your feedback.

**Comment by duncanokeyo** [comment]
thanks,,,  the library is becoming more stable,  will report more issues..
